conan/tools/premake/toolchain.py

```diff
@@ -0,0 +1,245 @@
+import os
+import textwrap
+from pathlib import Path
+
+from jinja2 import Environment
+
+from conan.tools.build.cross_building import cross_building
+from conan.tools.files import save
+from conan.tools.microsoft.visual import VCVars
+from conan.tools.premake.premakedeps import PREMAKE_ROOT_FILE
+
+_jinja_env = Environment(trim_blocks=True, lstrip_blocks=True)
+
+def _generate_flags(self, conanfile):
+    template = _jinja_env.from_string(textwrap.dedent(
+        """\
+        -- Workspace flags
+
+        {% if extra_cflags %}
+        -- C flags retrieved from CFLAGS environment, conan.conf(tools.build:cflags) and extra_cflags
+        filter {"files:**.c"}
+            buildoptions { {{ extra_cflags }} }
+        filter {}
+        {% endif %}
+        {% if extra_cxxflags %}
+        -- CXX flags retrieved from CXXFLAGS environment, conan.conf(tools.build:cxxflags) and extra_cxxflags
+        filter {"files:**.cpp", "**.cxx", "**.cc"}
+            buildoptions { {{ extra_cxxflags }} }
+        filter {}
+        {% endif %}
+        {% if extra_ldflags %}
+        -- Link flags retrieved from LDFLAGS environment, conan.conf(tools.build:sharedlinkflags), conan.conf(tools.build:exelinkflags) and extra_cxxflags
+        linkoptions { {{ extra_ldflags }} }
+        {% endif %}
+        {% if extra_defines %}
+        -- Defines retrieved from DEFINES environment, conan.conf(tools.build:defines) and extra_defines
+        defines { {{ extra_defines }} }
+        {% endif %}
+    """))
+
+    def format_list(items):
+        return ", ".join(f'"{item}"' for item in items) if items else None
+
+
+    build_env = self._conanfile.buildenv.vars(self._conanfile)
+    def _get_env_list(env):
+        v = build_env.get(env, [])
+        return v.strip().split() if not isinstance(v, list) else v
+
+    extra_defines = format_list(
+        _get_env_list("DEFINES")
+        + conanfile.conf.get("tools.build:defines", default=[], check_type=list)
+        + self.extra_defines
+    )
+    extra_c_flags = format_list(
+        _get_env_list("CFLAGS")
+        + conanfile.conf.get("tools.build:cflags", default=[], check_type=list)
+        + self.extra_cflags
+    )
+    extra_cxx_flags = format_list(
+        _get_env_list("CXXFLAGS")
+        + conanfile.conf.get("tools.build:cxxflags", default=[], check_type=list)
+        + self.extra_cxxflags
+    )
+    extra_ld_flags = format_list(
```

Comment:
Is it really necessary to read from Enviroment the CXXFLAGS and others and inject them?
Other generators such as ``CMakeToolchain`` or ``MSBuildToolchain`` don't do it, I am not sure this should be done. I know Meson does it, but it is the exception.

Reply:
I think it will provide extra flexibility to the users, maybe overkill(?)
It is already tested and works well.
Also, it will provide a similar behavior to `MesonToolchain` as you mentioned.

Reply:
The MesonToolchain is the exception, and it "slipped" there because of legacy reasons, the other generators don't provide it, Meson is the exception, and better not have extra exceptions.

Better not aim for "extra flexibility", but better aim for minimal, user-feedback driven features, otherwise it is too speculative, but adds complexity and maintenance to the code